src/main/scala/com/redhat/et/silex/util/sink.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * sink.scala
+ * 
+ * author:  William Benton <willb@redhat.com>
+ *
+ * Copyright (c) 2010-2016 Red Hat, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.et.silex.util
+
+/**
+ * On-line mean and variance estimates for a stream of Double values.
+ * Uses the technique from <a href="http://dl.acm.org/citation.cfm?id=359153">"Updating mean and variance estimates: an improved method"</a>, by D. H. D. West (1979).
+*/
+sealed class SampleSink(private var _count: Long, private var _min: Double, private var _max: Double, private var _mean: Double, private var _m2: Double) {
```

Comment:
Also a nit, but the name `SampleSink` left me confused about what it was doing.  Some name in the vein `IncrementalSufficientStats` or `IncrementalMoments`, etc...